src/db/drivers/mysql.ts

```diff
@@ -0,0 +1,219 @@
+        autoIncrement: true
+      },
+      groupName: {
+        type: new DataTypes.STRING(128),
+        allowNull: false,
+      }
+    }) as StaticSequelizeStudyGroupModel;
+  }
+
+  /**
+   * Confirms authentication with the database and sets up the required tables.
+   *
+   * @returns a promise that attempts to authenticate with the database and set up the tables
+   */
+  public setUp(): Promise<string> {
+    return new Promise((resolve, reject) => {
+      this.sequelize.authenticate()
+        // if we successfully authenticated with the database, set up our tables
+        .then(() => {
+          // if using a test database, drops any previously created tables
+          // TODO: look into using migration and the Sequelize CLI in production
+          this.sequelize.sync({ force: true, match: /_test$/ })
+            .then(() => resolve("Connected to MySQL database!"))
+            .catch(err => reject(`Model synchronization failed: ${err}`));
+        })
+        .catch(err => reject(`Failed to connect to MySQL database: ${err}`));
+    });
+  }
+
+  /**
+   * Gets a user from the database given their ID.
+   *
+   * @param id ID of the user to get
+   * @returns promise containing the desired user if resolved,
+   *          and an error message if rejected
+   */
+  public getUser(id: number): Promise<User> {
+    return new Promise((resolve, reject) => {
+      this.sequelizeUserModel.findByPk(id, { rejectOnEmpty: true })
+        .then(mySqlUser => {
+          const user = new User(mySqlUser.id, mySqlUser.username);
+          resolve(user);
+        })
+        .catch(err => reject(`Failed to get user: ${err}`));
+    });
+  }
+
+  /**
+   * Adds a user to the users table.
+   *
+   * @param username username of the user to create
+   * @returns promise containing newly created user if resolved,
+   *          and an error message if rejected
+   */
+  public createUser(username: string): Promise<User> {
+    return new Promise((resolve, reject) => {
+      this.sequelizeUserModel.create({ username: username })
+        .then(mySqlNewUser => {
+          const newUser = new User(mySqlNewUser.id, mySqlNewUser.username);
+          resolve(newUser);
+        })
+        .catch(err => reject(`Failed to create user: ${err}`));
+    });
+  }
+
+  /**
+   * Delete a user from our users table.
+   *
+   * @param id ID of the user to delete
+   * @returns promise that resolves with true if a user was deleted and false
+   *          if that user cannot be found, and rejects with an error message
+   */
+  public deleteUser(id: number): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      this.sequelizeUserModel.destroy({ where: { id: id } })
+        .then(rowsDeleted => {
+          // assuming it is impossible for rowsDeleted to be greater than 1, because if
+          // we're autoincrementing IDs we should be deleting at most one user
+          if (rowsDeleted < 1) {

```

Comment:
Shouldn't rowsDeleted be either 1 or 0? Why are you testing for any integer < 1? If a -1 is possible for error in the destroy function, shouldn't that be treated differently?

Reply:
Used a different implementation so as to avoid this issue.